### PR TITLE
Add guides

### PR DESCRIPTION
**Pull request by hassox**
I think docs need to be generated with 1.6. It seems that there has been a change to the docs generator.

### CODE REVIEW

File: guides/tutorial/start-tutorial.md [hunk 1]
```diff
@@ -1,3 +1,384 @@
...
+
+You should change the secret key for each environment and manage them with your secret management strategy.
+
+## Password hashing
+
+This too is not strictly required for Guardian. If you already have a way for you to verify user/password for your user model you can skip this part.
+
+We'll implement a simple version of password hashing for this tutorial. This is up to your application and is only shown here for example purposes.
+
+We added `:comonin` and `:bcrypt_elixir` to our mix deps at the start. We're going to use them in two places.
```
Comment (scrogson) [inline guides/tutorial/start-tutorial.md]: `comeonin`

File: guides/tutorial/start-tutorial.md [hunk 2]
```diff
@@ -1,3 +1,384 @@
...
+alias Comeonin.Bcrypt
+
+def changeset(%User{} = user, attrs) do
+  user
+  |> cast(attrs, [:username, :password])
+  |> validate_required([:username, :password])
+  |> put_password_hash()
+end
+
+defp put_password_hash(%Ecto.Changeset{valid?: true, changes: %{password: password}} = cset) do
```
Comment (scrogson) [inline guides/tutorial/start-tutorial.md]: I think we should use `changeset` in full just to make it clearer.

File: guides/tutorial/start-tutorial.md [hunk 3]
```diff
@@ -1,3 +1,384 @@
-# not yet implemented
+# Getting started with Guardian
 
-if you'd like to help with this guide please submit a pull request to (guardian)[https://github.com/ueberauth/guardian] in the `guides/tutorial/start-turorial.md` file.
+Getting started with Guaridan is easy. This tutorial will cover
+
+* Setting up the basics
```
Comment (scrogson) [inline guides/tutorial/start-tutorial.md]: These bullets are not all that clear...what does "Setting up the basics" mean?

"HTTP based setup" is also confusing to me. The word "setup" seems to be what we did in the first bullet point?

File: guides/tutorial/start-tutorial.md [hunk 4]
```diff
@@ -1,3 +1,384 @@
...
+* HTTP based setup
+* Login/Logout
+
+We'll use Phoenix for this tutorial since most folks will be using it. There is no requirement to use Phoenix with Guardian but it makes this tutorial easier.
+
+This tutorial was based on [this article](https://medium.com/@tylerpachal/session-authentication-example-for-phoenix-1-3-using-guardian-1-0-beta-a228c78478e6) by [Tyler Pachal](https://github.com/TylerPachal).
+
+We'll also use the default token type of JWT. Again with this you don't _have_ to use JWT for your token backend. See the [token documentation](tokens-start.html) for more information.
+
+Authentication consists of a challenge phase (prove who you are) and then followed by a verification phase (has this actor proven who they are?). Guardian looks after the second part for you. It's up to your application to implement the challenge phase after which Guardian will do the rest. In this tutorial we'll use [comonin](https://github.com/riverrun/comeonin) with [bcrypt](https://en.wikipedia.org/wiki/Bcrypt) for the challenge phase.
```
Comment (scrogson) [inline guides/tutorial/start-tutorial.md]: `comeonin`

File: guides/tutorial/start-tutorial.md [hunk 5]
```diff
@@ -1,3 +1,384 @@
...
+defp deps do
+  [
+    {:guardian, "~> 1.0"},
+    {:comeonin, "~> 4.0"},
+    {:bcrypt_elixir, "~> 0.12"},
+  ]
+end
+```
+
+## Create a user model
```
Comment (scrogson) [inline guides/tutorial/start-tutorial.md]: Maybe... `Create a user manager`

File: guides/tutorial/start-tutorial.md [hunk 6]
```diff
@@ -1,3 +1,384 @@
...
+
+We'll need something to authenticate. How Users are created and what they can do is outside the scope of this tutorial. If you already have a user model you can skip this part.
+
+```sh
+$ mix phx.gen.context Auth User users username:string password:string
```
Comment (scrogson) [inline guides/tutorial/start-tutorial.md]:
```
mix phx.gen.context UserManager User users username:string password:string
```

**Comment by hassox** [comment]
@scrogson updated

**Comment by hassox** [comment]
@scrogson ping

**Comment by scrogson** [comment]
@hassox I'll have a look a bit later this afternoon. Looking good mate!

**Pull request merged**